arviz/utils/xarray_utils.py

```diff
@@ -53,15 +55,156 @@ def __init__(self, obj, coords=None, dims=None):
     def to_xarray(self):
         pass
 
-    @abstractstaticmethod
-    def default_varnames_coords_dims(obj, coords, dims):  # pylint: disable=unused-argument
+    @abstractmethod
+    def default_varnames_coords_dims(self, obj, coords, dims):  # pylint: disable=unused-argument
         return
 
     @abstractmethod
     def verify_coords_dims(self):
         return
 
 
+class DictToXarray(Converter):
+    def __init__(self, obj, coords=None, dims=None, chains=None):
+        """Convert a dict to an xarray dataset.
+
+        Parameters
+        ----------
+        obj : dict[str, iterable]
```

Comment:
What do people think of using type annotations?

Reply:
I have no strong feelings

Reply:
yes, it can be gradually introduced, too.  i don't use pycharm, but i've heard it is quite helpful there.  i could set up a CI check if someone started adding type annotations (doesn't have to be here, though).

I'd guess it was difficult for some of these "edge" functions that accept objects that `ArviZ` does not know about (neither PyStan nor PyMC3 are requirements).